sources/packages/backend/apps/db-migrations/src/sql/Applications/Add-application-edit-status-columns.sql

```diff
@@ -0,0 +1,57 @@
+ALTER TABLE
+  sims.applications
+ADD
+  -- Adding new column with the default 'Edited' to later be removed.
+  COLUMN application_edit_status sims.application_edit_status NOT NULL DEFAULT 'Edited',
+ADD
+  -- Default will be dropped at the end of this script.
+  COLUMN application_edit_status_updated_on TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+ADD
+  -- A NOT NULL constraint will be added at the end of this script.
+  COLUMN application_edit_status_updated_by INT REFERENCES sims.users(id),
+ADD
+  COLUMN application_edit_status_note_id INT REFERENCES sims.notes(id);
+
+COMMENT ON COLUMN sims.applications.application_edit_status IS 'Status of the application related to possible editions that resulted in different versions of the same application.';
+
+COMMENT ON COLUMN sims.applications.application_edit_status_updated_on IS 'Last date and time when the edit status changed.';
+
+COMMENT ON COLUMN sims.applications.application_edit_status_updated_by IS 'User that changed the edit status last time.';
+
+COMMENT ON COLUMN sims.applications.application_edit_status_note_id IS 'Note added by the Ministry while approving or declining the edited application.';
+
+-- First ever saved application should be considered the original one.
+-- The submission will make the parent_application_id and the preceding_application_id the same from its own ID.
+UPDATE
+  sims.applications
+SET
+  application_edit_status = 'Original'
+WHERE
+  id = parent_application_id
+  AND parent_application_id = preceding_application_id;
+
+-- Consider the student as the one updating the application_edit_status_updated_by because
+-- at this moment only students can edit the applications and no Ministry approval is required.
+UPDATE
```

Comment:
For the optics of production data, would it not be beneficial to update the `application_status_updated_on` where the application status = Edited ?

Reply:
Changed as above.

Reply:
Thanks for the clarification. I updated as we talked.

Reply:
Just having second thoughts about how to have it more accurate.
The `application_status_updated_on` date evolves differently than the `application_edit_status_updated_on`.
I believe that we can update it as below.
- For `application_edit_status_updated_on` as `Original` we can use the creation date, since it will be set during the `draft` creation.
- For `application_edit_status_updated_on` as `Edited` we can use the `submittedDate`.

Reply:
Below are the suggested dates.
```sql
-- By the default constraint, all applications are considered "Original".
-- Set the update date to be the same as the creation date, since it will be updated when the draft application is created.
UPDATE
  sims.applications
SET
  application_edit_status_updated_on = created_at
WHERE
  id == parent_application_id;

-- Edited applications will be created during submit only for now, once the
-- application_edit_status_updated_on can be considered the same as the submitted_date.
UPDATE
  sims.applications
SET
  application_edit_status = 'Edited',
  application_edit_status_updated_on = submitted_date
WHERE
  id != parent_application_id;
```